src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -213,6 +213,24 @@ public ConfidentialClientApplicationBuilder WithAzureRegion(string azureRegion =
             return this;
         }
 
+
+        /// <summary>
+        /// When set to <c>true</c>, MSAL will lock cache access at the <see cref="ConfidentialClientApplication"/> level, i.e.
+        /// the block of code between BeforeAccessAsync and AfterAccessAsync callbacks will be syncronized. 
+        /// Apps can set this flag to <c>false</c> to enable an optimistic cache locking strategy, which may result in better performance, especially 
+        /// when ConfidentialClientApplication objects are reused.
+        /// </summary>
+        /// <remarks>
+        /// True by default, but subject to change.
+        /// Not recommended for apps that call RemoveAsync
```

Comment:
why @bgavrilMS ?

Reply:
I think that with optimistic synchronization there is a possibility that some updates will be ignored. If the update is "add a new token to the cache" and is ignored, it's not a big problem. But remove account is pretty bad. 

After we add a partitioned cache to ALL cca flows, this will no longer be a problem.